### DISCUSSION TITLE
Use one backbone with different heads - yolov5s

### DISCUSSION THREAD

**Discussion by brunopatricio2012** [ultralytics/yolov5#8746] [Q&A] [answered]
I am using the yolov5s model to do my training. I have 5 heads and I would like to make the training time easier, because I am training thousands of images and every time I want to add images to the model, I have to train everything again, is there any way to modify the yolov5s model the backbone, to add images to what is already trained? Is it possible to run the backbone only once with for example 3 heads? 

Thanks for your help

**Comment by glenn-jocher** [comment, marked as answer]
@brunopatricio2012 not sure what you mean by 5 heads, but no there is no easy way to simply 'add' new images to a model without it forgetting old ones. 

You can freeze layers easily though during training:
```
python train.py --freeze 10  # freeze first 10 layers (all of backbone)
```
See --freeze argument for details
https://github.com/ultralytics/yolov5/blob/3e858633b283767f038b4cab910a95e40fe8577b/train.py#L480

https://github.com/ultralytics/yolov5/blob/3e858633b283767f038b4cab910a95e40fe8577b/train.py#L130-L137

**Comment by brunopatricio2012** [comment]
Ok, thanks for the answer.
What I intend to do is this, represented in the figure below.

![example](https://user-images.githubusercontent.com/24433634/181301444-5e5549de-675d-4037-b1f2-5438771d4ac3.jpg)

Thanks again!
